x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeFieldIndexShadowTests.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.runtimefields.mapper;
+
+import org.elasticsearch.index.mapper.DocumentMapper;
+import org.elasticsearch.index.mapper.MapperServiceTestCase;
+import org.elasticsearch.index.mapper.ParsedDocument;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.xpack.runtimefields.RuntimeFields;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+
+public class RuntimeFieldIndexShadowTests extends MapperServiceTestCase {
+
+    @Override
+    protected Collection<? extends Plugin> getPlugins() {
+        return Collections.singleton(new RuntimeFields());
+    }
+
+    public void testRuntimeFieldShadowsObject() throws IOException {
```

Comment:
could this test fit into the existing DocumentParserTests? I have added similar tests in there as part of #65210